File: backend/src/main/java/com/cdac/farmermarketplace/service/CartService.java
package com.cdac.farmermarketplace.service;


import com.cdac.farmermarketplace.dto.request.AddToCartRequest;
import com.cdac.farmermarketplace.dto.request.UpdateCartRequest;
import com.cdac.farmermarketplace.dto.response.CartResponse;



public interface CartService {



    // Add product to logged-in user's cart
    CartResponse addToCart(
            AddToCartRequest request,
            Long userId
    );




    // Get cart of logged-in user
    CartResponse getCartByUserId(
            Long userId
    );




    // Update quantity
    CartResponse updateCart(
            UpdateCartRequest request,
            Long userId
    );




    // Remove item from cart
    void removeCartItem(
            Long cartItemId,
            Long userId
    );




    // Clear complete cart
    void clearCart(
            Long userId
    );



}

File: frontend/src/services/cartService.js
import api from "./api";

const cartService = {

  // Get Logged-in User Cart
  getCart: async () => {
    const response = await api.get("/cart");
    return response.data;
  },

  // Add Product To Cart
  addToCart: async (productId, quantity) => {
    const response = await api.post("/cart/add", {
      productId,
      quantity,
    });

    return response.data;
  },

  // Update Quantity
  updateCart: async (cartItemId, quantity) => {
    const response = await api.put("/cart/update", {
      cartItemId,
      quantity,
    });

    return response.data;
  },

  // Remove Item
  removeCartItem: async (cartItemId) => {
    const response = await api.delete(`/cart/item/${cartItemId}`);
    return response.data;
  },

  // Clear Cart
  clearCart: async () => {
    const response = await api.delete("/cart/clear");
    return response.data;
  },
};

export default cartService;